sentinel/lib/messages.js

```diff
@@ -175,12 +176,21 @@ module.exports = {
             phone = utils.getHealthCenterPhone(doc);
         } else if (recipient === 'grandparent') {
             phone = utils.getDistrictPhone(doc);
-        }
-
-        if (!phone && doc.fields && doc.fields[recipient]) {
-            // try to resolve a specified property/field name
+        } else if (doc.fields && doc.fields[recipient]) {
+            // Try to resolve a specified property/field name
             phone = doc.fields[recipient];
+        } else if (doc[recipient]) {
+            // Or directly on the doc
+            phone = doc[recipient];
+        } else if (recipient.indexOf('.') > -1) {
+            // Or multiple layers by executing it as a statement
+            try {
+                phone = objectPath.get(doc, recipient);
+            } catch (err) {
+                logger.error(`Recipient expression "${recipient}" failed on ${doc._id}`);
+            }
```

Comment:
I don't think objectpath ever throws (although you should confirm) so we don't need the try/catch any more.

Reply:
In classic new-age documentation style it never mentions edge cases anywhere, so I can't be sure without walking through the code. However, I can't *make* it crash, so I guess that's good enough?